File: it.unibo.bls19Local/src/main/java/it/unibo/bls/devices/gui/ButtonBasic.java
package it.unibo.bls.devices.gui;
import java.awt.BorderLayout;
import java.awt.Frame;
import java.awt.event.ActionListener;
import it.unibo.bls.interfaces.IButton;

public class ButtonBasic extends java.awt.Button {
	public ButtonBasic(Frame frame, String label, ActionListener listener){
		super(label);
		this.addActionListener(  listener );
		frame.add(BorderLayout.WEST,this); 
		frame.validate();
	}
}

File: it.unibo.bls19d/src/main/kotlin/it/unibo/bls19d/qak/ButtonBasic.kt
package it.unibo.bls19d.qak

import java.awt.*
import java.awt.event.ActionListener

class ButtonBasic(frame: Frame, label: String, listener: ActionListener) : java.awt.Button(label) {
    init {
        this.addActionListener(listener)
        frame.add(BorderLayout.WEST, this)
        frame.validate()
    }
}